### PR TITLE
Patch react-autosize-textarea for updated types

### PR DESCRIPTION
**Pull request by sirreal**
## What?

Patch `react-autosize-textarea` to fix some type issues. This should be a harmless change.

There are bigger problems with the library that can be addressed, see #61568.

This is a short term fix that should be harmless and fix some build issues.

Extracted from #61486

## Why?

Incompatible types can cause this to break because there are type errors between the library and recent React types.

## How?

Patch the package to remove references to some keys that no longer exist in the React types.

## Testing Instructions

On install you'll see the following (note `react-autosize-textarea@7.1.0 ✔`):

```
> gutenberg@18.3.0 postinstall
> patch-package && node ./patches/patch-xcode.js

patch-package 8.0.0
Applying patches...
lighthouse@10.4.0 ✔
react-autosize-textarea@7.1.0 ✔
react-devtools-core@4.28.5 ✔
react-native@0.73.3 ✔
```

CI is sufficient for testing.

### CODE REVIEW

**Review by sirreal — COMMENTED** [review]
This removes `onPointerEnterCapture"` and `onPointerLeaveCapture`, which were removed from React.

https://github.com/facebook/react/issues/17883

File: patches/react-autosize-textarea+7.1.0.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/react-autosize-textarea/lib/TextareaAutosize.d.ts b/node_modules/react-autosize-textarea/lib/TextareaAutosize.d.ts
```
Comment (youknowriad) [inline patches/react-autosize-textarea+7.1.0.patch:1]: So by adding a patches file like that, npm automatically overrides the dependency or something? How does that work. TIL

Comment (youknowriad) [reply patches/react-autosize-textarea+7.1.0.patch:1]: Also is there a way to add a comment or something to explain why we need the patch? Git blame might suffice though.

Comment (sirreal) [reply patches/react-autosize-textarea+7.1.0.patch:1]: Yarn has this functionality built in. Gutenberg uses the `patch-package` package and runs it on postinstall to do the patching:

https://github.com/WordPress/gutenberg/blob/b83f0f983d8da9a6709c4237281889b07158da4b/package.json#L311

I'll add a README to the patches dir, that's a good plan.

**Review by youknowriad — APPROVED** [review]
The README is very useful. Thank you.

**Pull request merged**